frontend/subscriptions/index.js

```diff
@@ -94,4 +102,12 @@ export default (subscribe) => {
         });
     }
   });
+
+  subscribe(productsAdded$, ({ dispatch }) => {
+    dispatch(setPauseRechargeCart(true));
+  });
+
+  subscribe(productsUpdated$, ({ dispatch }) => {
+    dispatch(setPauseRechargeCart(false));
```

Comment:
have you tested what happens if the add product fails?

Reply:
Tested and item does not get added to cart. No future blockage.